app/src/main/java/com/realityexpander/tasky/agenda_feature/presentation/add_event_screen/AddEventViewModel.kt

```diff
@@ -0,0 +1,101 @@
+package com.realityexpander.tasky.agenda_feature.presentation.add_event_screen
+
+import androidx.compose.ui.tooling.preview.datasource.LoremIpsum
+import androidx.lifecycle.SavedStateHandle
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.realityexpander.tasky.R
+import com.realityexpander.tasky.agenda_feature.domain.IAgendaRepository
+import com.realityexpander.tasky.agenda_feature.presentation.agenda_screen.AgendaEvent
+import com.realityexpander.tasky.auth_feature.domain.AuthInfo
+import com.realityexpander.tasky.auth_feature.domain.IAuthRepository
+import com.realityexpander.tasky.core.presentation.common.SavedStateConstants
+import com.realityexpander.tasky.core.presentation.common.util.UiText
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.*
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.yield
+import java.time.ZonedDateTime
+import java.time.temporal.ChronoUnit
+import javax.inject.Inject
+
+@HiltViewModel
+class AddEventViewModel @Inject constructor(
+    private val authRepository: IAuthRepository,
+    private val agendaRepository: IAgendaRepository,
+    private val savedStateHandle: SavedStateHandle,
+) : ViewModel() {
+
+    // Get params from savedStateHandle (from another screen or after process death)
+    private val errorMessage: UiText? =
+        savedStateHandle[SavedStateConstants.SAVED_STATE_errorMessage]
+
+    private val _addEventState = MutableStateFlow(AddEventState())
+    val addEventState = _addEventState.onEach { state ->
+        // save state for process death
+        savedStateHandle[SavedStateConstants.SAVED_STATE_errorMessage] = state.errorMessage
+    }.stateIn(viewModelScope, SharingStarted.WhileSubscribed(5000), AddEventState())
+
+    init {
+        viewModelScope.launch {
+            // restore state after process death
+            _addEventState.update {
```

Comment:
You don't need this `update` in `init`, since you can just change the default value of your state in `MutableStateFlow(defaultValue)`. That way you don't need the copy and update operation

Reply:
Ah ok, that makes sense. I will move this to the "initial state" of the  MutableStateFlow of the state